### PR TITLE
feature: support show database rule from

### LINKED ISSUE #627: [SQL] Support `show database rules from xxx`
**What would you like to be added**:

**Why is this needed**:

`show database rules from student;`

- Task 1: SQL parser support this sql statement, reference: https://github.com/arana-db/parser/pull/9
- Task 2: Support sql statement at `arana` project, reference: https://github.com/arana-db/arana/pull/284

### PR DESCRIPTION
**Pull request by Lvnszn**
**What this PR does**:
support `show database rule from`

**Which issue(s) this PR fixes**:

Fixes #627

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/runtime/plan/dal/show_database_rules_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dal
+
+import (
+	"github.com/arana-db/parser"
```
Comment (Mulavar) [inline pkg/runtime/plan/dal/show_database_rules_test.go:21]: format it with imports-formatter.

File: test/integration_test.go [hunk 2]
```diff
@@ -644,6 +644,19 @@ func (s *IntegrationSuite) TestShowShardingTable() {
 	assert.NoErrorf(t, err, "show sharding table from employees error: %v", err)
 }
 
+func (s *IntegrationSuite) TestShowDatabaseRules() {
+	var (
+		db = s.DB()
+		t  = s.T()
+	)
+
+	rows, err := db.Query("show database rules from employees")
```
Comment (Mulavar) [inline test/integration_test.go:653]: should we construct some mock rule to validate the result set?

**Review by maronghe — APPROVED** [review]

File: test/integration_test.go (hunk 2 above)
Comment (Lvnszn) [reply test/integration_test.go:653]: Need to add db rule, which may affect other ut

File: pkg/runtime/plan/dal/show_database_rules_test.go (hunk 1 above)
Comment (Lvnszn) [reply pkg/runtime/plan/dal/show_database_rules_test.go:25]: addressed

**Comment by jjeffcaii** [comment]
I think the `DBRule` is existing in the `VTable`, please check the `VTable` and adjust your codes. Thanks.

**Comment by Lvnszn** [comment]
> I think the `DBRule` is existing in the `VTable`, please check the `VTable` and adjust your codes. Thanks.

addressed

**Comment by Lvnszn** [comment]
It should be noted that this function will return all db rules in the database.

**Comment by jjeffcaii** [comment]
> It should be noted that this function will return all table‘s db rules in the database.

In my point of view, it should mean `show database rule from <some_table>`, and will return the db sharding rule of the target table.

**Comment by Lvnszn** [comment]
> > It should be noted that this function will return all table‘s db rules in the database.
> 
> In my point of view, it should mean `show database rule from <some_table>`, and will return the db sharding rule of the target table.

If there are the same table names in different databases, if there is no database identifier, this query will get tables with the same name, but they say they come from different databases. Does this logic need to be adjusted?

**Review by jjeffcaii — APPROVED** [review]

**Review by Mulavar — APPROVED** [review]

**Pull request merged**